src/server/server.py

```diff
@@ -1,25 +1,138 @@
+import logging
 import grpc
 from concurrent import futures
-from src.generated.search_pb2 import PongResponse
-from src.generated.search_pb2_grpc import PingPongServicer, add_PingPongServicer_to_server
+from src.server.elasticsearch_client import ElasticsearchClient
+
+from grpc_reflection.v1alpha import reflection
+from src.generated import search_pb2, search_pb2_grpc, recipe_search_service_pb2, recipe_search_service_pb2_grpc
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    handlers=[logging.StreamHandler()],
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PingPongService(search_pb2_grpc.PingPongServicer):
+    """
+    Service for handling ping-pong requests.
+    """
 
-class PingPongService(PingPongServicer):
     def Ping(self, request, context):
-        return PongResponse(message=f"Pong: {request.message}")
+        """
+        Handles a ping request and returns a pong response.
+        
+        Args:
+            request: The ping request.
+            context: The gRPC context.
+        
+        Returns:
+            A pong response.
+        """
+        return search_pb2.PongResponse(message=f"Pong: if I change here Ping should Change {request.message}")
+
+class RecipeSearchService(recipe_search_service_pb2_grpc.RecipeSearchServiceServicer):
+    """
+    Service for handling search recipe
+    """
+    def __init__(self):
+        # Initialize Elasticsearch client
+        self.es_client = ElasticsearchClient()
+
+    def SearchRecipes(self, request, context):
```

Comment:
```suggestion
    def search_recipes(self, request, context):
```
and please add typing